week-1-ts-ecosystem/w01e03-object-oriented-structure/src/controllers/Cart.ts

```diff
@@ -0,0 +1,53 @@
+import { Item } from "../models/Item";
+
+export class Cart<Product extends Item> {
+  private producs: Map<string, Product> = new Map();
+
+  add(product: Product) {
+    this.producs.set(product.id, product);
+  }
+
+  findById(id: string) {
+    return this.producs.get(id);
+  }
+
+  updateById(id: string, product: Product) {
+    if (!!this.producs.get(id)) {
+      product.id = id;
+      this.producs.set(id, product);
+      return;
+    }
+    console.log("Nie znaleziono produktu o podanym id");
+  }
+
+  removeById(id: string) {
+    this.producs.delete(id);
+  }
+
+  getAll() {
+    return [...this.producs.values()];
+  }
+
+  getCount() {
+    return this.producs.size;
+  }
+
+  getAmount() {
+    return [...this.producs].reduce((acc, [, product]) => acc + product.amount, 0);
```

Comment:
Czemu tutaj taka komplikacja zamiast `[...this.products.values()]` ?
Albo lepiej żeby sie nie powtarzać:

`this.getAll().reduce .... `